### PR TITLE
HPCC4J-394 Synchronize with ESP 8.0.x

### PR DESCRIPTION
**Pull request by rpastrana**
- Adds WsCodeSign client 1.01
- Adds WsResources client 1.00
- Updates WsWorkunits 1.81, WsDFU 1.57

Signed-off-by: Rodrigo Pastrana <Rodrigo.Pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsCodeSignClient.java [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+                initErrMessage += "\n" + e.getLocalizedMessage();
+            }
+            log.error(initErrMessage);
+        }
+    }
+
+    public ListUserIDsResponseWrapper listUserIDs(ListUserIDsRequestWrapper req) throws Exception
+    {
+        if (req == null)
+            throw new Exception("");
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsCodeSignClient.java:211]: Need an exception message

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsCodeSignClient.java [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+            handleEspExceptions(new ArrayOfEspExceptionWrapper(exceptions), "Error listing userids");
+        }
+
+        return new ListUserIDsResponseWrapper(resp);
+    }
+
+    public SignResponseWrapper sign(SignRequestWrapper req) throws Exception
+    {
+        if (req == null)
+            throw new Exception("");
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsCodeSignClient.java:238]: Need an exception message

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsCodeSignClient.java [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+            handleEspExceptions(new ArrayOfEspExceptionWrapper(exceptions), "Error signing");
+        }
+
+        return new SignResponseWrapper(resp);
+    }
+
+    public VerifyResponseWrapper verify(VerifyRequestWrapper req) throws Exception
+    {
+        if (req == null)
+            throw new Exception("");
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsCodeSignClient.java:265]: Missing exception message

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsResourcesClient.java [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+     * Submit service query request
+     *
+     * @param req
+     * @return
+     * @throws Exception
+     */
+    public ServiceQueryResponseWrapper serviceQuery(ServiceQueryRequestWrapper req) throws Exception
+    {
+        if (req == null)
+            throw new Exception("");
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsResourcesClient.java:207]: Missing exception message

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsResourcesClient.java [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+    {
+        verifyStub();
+
+        try
+        {
+            ((WsResourcesStub) stub).ping(new WsResourcesPingRequest());
+        }
+        catch (Exception e)
+        {
+            return false;
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsResourcesClient.java:247]: Should we be letting the exception get thrown here?

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsSMCClient.java [hunk 6]
```diff
@@ -219,6 +227,80 @@ public String getHPCCBuild() throws Exception
...
+           log.error("HPCCWsSMCClient.isContainerized encountered RemoteException.\n" + e.getLocalizedMessage());
+       }
+
+       if (resp != null)
+       {
+           if (resp.getExceptions() != null)
+               handleEspExceptions(new ArrayOfEspExceptionWrapper(resp.getExceptions()), "Could not query container mode info");
+       }
+
+       ArrayOfNamedValue buildinfo = resp.getBuildInfo();
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsSMCClient.java:258]: Should handle resp == null before this?

**Review by jpmcmu — COMMENTED** [review]
@rpastrana still reviewing. Initial set of comments:

**Comment by rpastrana** [comment]
@jpmcmu please review

File: wsclient/src/main/java/org/hpccsystems/ws/client/wrappers/wsworkunits/WUSubmitWrapper.java [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    private Integer maxruntime = null;
+    private Integer blocktillfinishtimer = null;
+    private Integer maxcost = null;
+    private Boolean notifycluster = null;
+    private String queue = null;
+    private Boolean syntaxcheck = null;
+
+    public String getWuid()
+    {
+		return wuid;
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/wrappers/wsworkunits/WUSubmitWrapper.java:18]: Formatting looks off here

File: wsclient/src/main/java/org/hpccsystems/ws/client/wrappers/wsworkunits/WUSubmitWrapper.java [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+	}
+
+    /**
+     * Gets the raw version 1_81.
+     *
+     * @return the raw version 1_81
+     */
+    public org.hpccsystems.ws.client.gen.axis2.wsworkunits.v1_81.WUSubmit getRaw()
+    {
+    	WUSubmit request = new org.hpccsystems.ws.client.gen.axis2.wsworkunits.v1_81.WUSubmit();
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/wrappers/wsworkunits/WUSubmitWrapper.java:143]: Formatting

File: wsclient/src/test/java/org/hpccsystems/ws/client/WSCodeSignClientTest.java [hunk 9]
```diff
@@ -0,0 +1,188 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2019 HPCC Systems®.
```
Comment (jpmcmu) [inline wsclient/src/test/java/org/hpccsystems/ws/client/WSCodeSignClientTest.java:3]: Change date

File: wsclient/src/test/java/org/hpccsystems/ws/client/WSResroucesClientTest.java [hunk 10]
```diff
@@ -0,0 +1,114 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2019 HPCC Systems®.
```
Comment (jpmcmu) [inline wsclient/src/test/java/org/hpccsystems/ws/client/WSResroucesClientTest.java:3]: Date

**Review by jpmcmu — COMMENTED** [review]
@rpastrana a few more minor comments

**Review by jpmcmu — APPROVED** [review]
@rpastrana looks good to me

**Pull request merged**